src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -63,14 +63,100 @@ internal static IEnumerable<RuntimeLibrary> GetCandidateLibraries(DependencyCont
                 return Enumerable.Empty<RuntimeLibrary>();
             }
 
-            return dependencyContext.RuntimeLibraries.Where(IsCandidateLibrary);
+            var candidatesResolver = new CandidatesResolver(dependencyContext.RuntimeLibraries, ReferenceAssemblies);
+            return candidatesResolver.GetCandidates();
         }
 
-        private static bool IsCandidateLibrary(RuntimeLibrary library)
+        private class CandidatesResolver
```

Comment:
This one wasn't a typo, it was Candidates on purpose, but I've changed it to CandidateResolver anyway